source/edutravel/travel_agency/models.py

```diff
@@ -252,14 +282,132 @@ class Student(Person):
+    name = models.CharField(max_length=80)
+    short_code = models.CharField(max_length=3)
+
+    def __str__(self):
+        return self.name
+
+class Country(models.Model):
+    '''
+    Contains the list of Counties
+    '''
+    name = models.CharField(max_length=80)
+    latitude = models.FloatField()
+    longitude = models.FloatField()
+    primary_language = models.ForeignKey(LanguageCode,on_delete=models.CASCADE)
+    capital = models.CharField(max_length=80)
+    hemisphere = models.CharField(max_length=8)
+
+    def __str__(self):
+        return self.name
+
+class Subdivision(models.Model):
+    country = models.ForeignKey(Country,on_delete=models.CASCADE)
+    latitude = models.FloatField()
+    longitude = models.FloatField()
+    name = models.CharField(max_length=80)
+
+    def __str__(self):
+        return self.name
+
+class City(models.Model):
+    subdivision = models.ForeignKey(Subdivision, on_delete=models.CASCADE)
+    latitude = models.FloatField()
+    longitude = models.FloatField()
+    Name = models.CharField()
+
+class TripCity(models.Model):
+    trip = models.ForeignKey(Trip,on_delete=models.CASCADE)
+    city = models.ForeignKey(City, on_delete=models.CASCADE)
+
+class ZipCode(models.Model):
+    zip_code = models.IntegerField()
+
+    def __str__(self):
+        return self.zip_code
+
+class TypeTable(models.Model):
+    name = models.CharField(max_length=80)
+    entity = models.CharField(max_length=80)
+
+    def __str__(self):
+        return self.name
+'''
+class LodgingType(models.Model):
+    merged into TypeTables
+'''
+
+class LocalLodging(models.Model):
+    city = models.ForeignKey(City, on_delete=models.CASCADE)
+    lodging_type = models.ForeignKey(TypeTables, on_delete=models.CASCADE, limit_choices_to={'entity': 'LocalLodging'})
+    street_address = models.CharField(max_length=80)
+    zip_code = models.ForeignKey(ZipCode, on_delete=models.CASCADE)
+    name = models.CharField(max_length=80)
+    rating = models.DecimalField(max_digits=2, decimal_places=1)
+
+    def __str__(self):
+        return self.name
+
+'''
+class BusinessType(models.Model):
+    merged into TypeTables
+'''
+class Industry(models.Model):
+    industry = models.CharField(max_length=80)
+
+    def __str__(self):
+        return self.industry
+
+class LocalBusiness(models.Model):
+    city = models.ForeignKey(City, on_delete=models.CASCADE)
+    business_type = models.ForeignKey(TypeTable, on_delete=models.CASCADE, limit_choices_to={'entity': 'LocalBusiness'})
+    name = models.CharField(max_length=80)
+    industry = models.ForeignKey(Industry,on_delete=models.CASCADE)
+    street_address = models.CharField(max_length=80)
+    zip_code = models.ForeignKey(ZipCode, on_delete=models.CASCADE)
+    is_active = models.BooleanField(default=true)
+
+
+
+class GuideSpecialty(models.Model):
+    specialty = models.CharField(max_length=80)
+
+    def __str__(self):
+        return self.specialty
+
+class LocalGuide(models.Model):
```

Comment:
Local guide can inherit from Person? This would lead to only having specialty and city attributes..